snuba/datasets/discover.py

```diff
@@ -197,19 +199,27 @@ def __init__(
     def select_storage(
         self, query: Query, request_settings: RequestSettings
     ) -> StorageAndMappers:
-        table = detect_table(
+        dataset = detect_dataset(
             query,
             self.__abstract_events_columns,
             self.__abstract_transactions_columns,
             True,
         )
-        return (
-            StorageAndMappers(self.__events_table, self.__event_translator)
-            if table == EVENTS
-            else StorageAndMappers(
+
+        if dataset == TRANSACTIONS:
+            return StorageAndMappers(
                 self.__transactions_table, self.__transaction_translator
             )
-        )
+        else:
+            use_readonly_storage = (
+                state.get_config("enable_events_readonly_table", False)
+                and not request_settings.get_consistent()
+            )
+            return (
+                StorageAndMappers(self.__events_ro_table, self.__event_translator)
+                if use_readonly_storage
```

Comment:
Won't this potentially select an empty table on development environments and on prem? None is writing on events_ro there. Or do you plan to address that subsequently making the events_ro table only be created in distributed environments ?
In such case will we even have a local table definition for this storage ?

Reply:
I don't think so. The local table name is the same in both storages: "sentry_local". The new storage picks the relevant name "sentry_local" or "sentry_dist_ro". It works the same way the old one picked between "sentry_local" and "sentry_dist".

Reply:
right.